### PR TITLE
add  releases-1.0.x  to the list of branches in GH actions

### PR DESCRIPTION
**Pull request by uturunku1**
## Description

Our github actions workflows are running for PRs that want to merge against main branch. We have a new branch "releases-1.0.x" for the purposes of releasing Go-TFE 1.0 and we should have the same github actions applied to this branch.

### CODE REVIEW

**Comment by uturunku1** [comment]
Am I smelling some flaky test?

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 name: golangci-lint
 on:
   push:
-    branches:
-      - main
+    branches: [ main, releases-1.0.x ]
   pull_request:
```
Comment (sebasslash) [inline .github/workflows/golangci-lint.yml:5]: So I'm actually curious about why the lint action fires on push and not when a PR is created? Should we move this to when a pull request is created to reduce the # of lint jobs fired?

**Review by sebasslash — COMMENTED** [review]
Everything else looks g2m

File: .github/workflows/golangci-lint.yml (hunk 1 above)
Comment (brandonc) [reply .github/workflows/golangci-lint.yml:5]: It probably needs to run on each commit?

Comment (uturunku1) [reply .github/workflows/golangci-lint.yml:5]: That's right. Push event is every commit that gets pushed. Pull request event is when you open a PR, when you synchronize a PR, when you close a PR, and a bunch of other activities that can happen for a PR. Though only some of those activities are enabled by default.

**Review by nfagerlund — APPROVED** [review]
Seems fine!

**Pull request merged**